core/controller/src/main/scala/whisk/core/controller/Rules.scala

```diff
@@ -396,13 +370,26 @@ trait WhiskRulesApi extends WhiskCollectionAPI {
      * @param action the action id
      * @return future that completes with references trigger and action if they exist
      */
-    private def checkTriggerAndActionExist(trigger: DocId, action: DocId)(implicit transid: TransactionId): Future[(WhiskTrigger, WhiskAction)] = {
+    private def checkTriggerAndActionExist(trigger: FullyQualifiedEntityName, action: FullyQualifiedEntityName)(
+        implicit transid: TransactionId): Future[(WhiskTrigger, WhiskAction)] = {
+
         for {
-            triggerExists <- WhiskTrigger.get(entityStore, trigger) recoverWith {
-                case _: NoDocumentException => Future.failed(new NoDocumentException(s"trigger $trigger does not exist"))
+            triggerExists <- WhiskTrigger.get(entityStore, trigger.toDocId) recoverWith {
+                case _: NoDocumentException => Future.failed {
+                    new NoDocumentException(s"trigger ${trigger.qualifiedNameWithLeadingSlash} does not exist")
+                }
+                case _: DeserializationException => Future.failed {
+                    new DeserializationException(s"trigger ${trigger.qualifiedNameWithLeadingSlash} is corrupted")
+                }
             }
-            actionExists <- WhiskAction.get(entityStore, action) recoverWith {
-                case _: NoDocumentException => Future.failed(new NoDocumentException(s"action $action does not exist"))
+
+            actionExists <- WhiskAction.get(entityStore, action.toDocId) recoverWith {
```

Comment:
short circuits the second fetch if it's not necessary - it's a CRUD, and on the success path, cache is prewarmed with these entities.